src/coreclr/jit/lclvars.cpp

```diff
@@ -2051,6 +2051,12 @@ unsigned Compiler::compMap2ILvarNum(unsigned varNum) const
         return (unsigned)ICorDebugInfo::RETBUF_ILNUM;
     }
 
+    if (info.compMethodInfo->args.hasImplicitThis() && varNum == info.compThisArg)
+    {
+        // implicit this doesn't map to any IL arg
```

Comment:
Implicit this maps to IL arg 0?

Reply:
Um.. does it? e.g. `comp->eeGetArgClass()` API doesn't return `this` for 0 arg and returns the actual 1st users parameter, I though that arg0 being `this` is only true for "explicit this"

Reply:
ah I see your comment in https://github.com/dotnet/runtime/pull/103503#discussion_r1641266868

Reply:
>  API doesn't return this for 0 arg and returns the actual 1st users parameter, I though that arg0 being this is only true for "explicit this"

Yes, this is true for signature walking APIs.